cmd/system-probe/modules/network_tracer.go

```diff
@@ -80,11 +103,86 @@ type networkTracer struct {
 	restartTimer *time.Timer
 }
 
+// startRPCServer is used to start a gRPC server using Unix sockets.
+func startRPCServer(tracer networkTracer, socketFilePath string) error {
+	// Remove existing socket file if it exists
+	os.Remove(socketFilePath)
+
+	listener, err := net.Listen("unix", socketFilePath)
+	if err != nil {
+		return err
+	}
+
+	server := grpc.NewServer()
+	connectionserver.RegisterSystemProbeServer(server, &tracer)
+
+	go func() {
+		log.Info("gRPC server listening on Unix domain socket...")
+
+		err = server.Serve(listener)
+		if err != nil {
+			log.Errorf("unable to serve gRPC server %v", err)
+		}
+	}()
+
+	// Wait for interrupt signal to gracefully stop the server
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM)
+	<-sigCh
```

Comment:
is that code exists with the http server as well?

Reply:
actually no, I removed it and validate that when I close the sp the server is closing as well